internal/extproc/watcher_test.go

```diff
@@ -33,6 +35,15 @@ func (m *mockReceiver) getConfig() *filterconfig.Config {
 	return m.cfg
 }
 
+// NewTestLoggerWithBuffer creates a new logger with a buffer for testing and asserting the output.
+func NewTestLoggerWithBuffer() (*slog.Logger, *bytes.Buffer) {
```

Comment:
do not need to be exported
```suggestion
// newTestLoggerWithBuffer creates a new logger with a buffer for testing and asserting the output.
func newTestLoggerWithBuffer() (*slog.Logger, *bytes.Buffer) {
```